uniswap/model/parts/uniswap_model.py

```diff
@@ -141,10 +141,32 @@ def s_mechanismHub_UNI(_params, substep, sH, s, _input):
 
 # AUX
 
-def getInputPrice(input_amount, input_reserve, output_reserve, _params):
+'''
+    Function to acquire amount of EHT or Token that a transaction will give.
+    input_amount Amount of ETH or Tokens being sold.
+    input_reserve Amount of ETH or Tokens (input type) in exchange reserves.
+    output_reserve Amount of ETH or Tokens (output type) in exchange reserves.
+    return: amount of ETH or Tokens bought.
+'''
+def getOutputAmount(input_amount, input_reserve, output_reserve, _params):
     fee_numerator = _params[0]['fee_numerator']
     fee_denominator = _params[0]['fee_denominator']
     input_amount_with_fee = input_amount * fee_numerator
     numerator = input_amount_with_fee * output_reserve
     denominator = (input_reserve * fee_denominator) + input_amount_with_fee
-    return int(numerator // denominator)
\ No newline at end of file
+    return int(numerator // denominator)
+
+'''
+    Function to adquire price of EHT or Token for a specific transaction.
+    input_amount Amount of ETH or Tokens being sold.
+    input_reserve Amount of ETH or Tokens (input type) in exchange reserves.
+    output_reserve Amount of ETH or Tokens (output type) in exchange reserves.
+    return: price amount of ETH or Tokens bought.
+'''
+def getExchangePrice(input_amount, input_reserve, output_reserve, _params):
+    fee_numerator = _params[0]['fee_numerator']
+    fee_denominator = _params[0]['fee_denominator']
+    input_amount_with_fee = input_amount * fee_numerator
+    numerator = input_amount_with_fee * output_reserve
+    denominator = (input_reserve * fee_denominator) + input_amount_with_fee
+    return input_amount  / (numerator // denominator)
```

Comment:
Looks like it works, but theoretically the input_amount just matters for the exchange price beucase of the fee, right? However, I tried to run the function without fee (fee_numerator = fee_denominator) and the price slightly changed if I changed the input_amount. Is there a reason why this should happen?

Reply:
Actually, I tested it again and understood what is happening. If input_amount receives a value that doesn't have 19 digits (its  smallest possible value), the function returns a distorted value. I don't think there's any situation where it wouldn't have 19 digits though, so I think we don't have to worry about it.